--- ggml-backend-reg.cpp.orig
+++ ggml-backend-reg.cpp
@@ -35,8 +35,11 @@
 #endif
 
 #ifdef LM_GGML_USE_METAL
+#include <TargetConditionals.h>
+#if !TARGET_OS_SIMULATOR
 #include "ggml-metal.h"
 #endif
+#endif
 
 #ifdef LM_GGML_USE_SYCL
 #include "ggml-sycl.h"
@@ -138,7 +141,9 @@
         register_backend(lm_ggml_backend_cuda_reg());
 #endif
 #ifdef LM_GGML_USE_METAL
-        register_backend(lm_ggml_backend_metal_reg());
+        #if !TARGET_OS_SIMULATOR
+            register_backend(lm_ggml_backend_metal_reg());
+        #endif
 #endif
 #ifdef LM_GGML_USE_SYCL
         register_backend(lm_ggml_backend_sycl_reg());
